java-checks/src/main/java/org/sonar/java/checks/UselessImportCheck.java

```diff
@@ -129,13 +130,29 @@ private void checkSymbolUsage(ImportTree importTree, String importName) {
       if (symbol != null) {
         Symbol owner = symbol.owner();
         // Exclude method symbols : they could be ambiguous or unresolved and lead to FP.
-        if (symbol.isVariableSymbol() && symbol.usages().stream().allMatch(identifierTree -> JUtils.enclosingClass(identifierTree) == owner)) {
+        if (symbol.isVariableSymbol() && symbol.usages().stream().allMatch(identifierTree -> enclosingClass(identifierTree) == owner)) {
           context.reportIssue(this, importTree, "Remove this unused import '" + importName + "'.");
         }
       }
     }
   }
 
+  private static Symbol enclosingClass(Tree t) {
```

Comment:
As it can return null (line 143), please annotated the method with `@CheckForNull`.